src/rnnlmbin/rnnlm-get-egs.cc

```diff
@@ -0,0 +1,159 @@
+// rnnlmbin/rnnlm-get-egs.cc
+
+// Copyright 2016  Hainan Xu
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include <fstream>
+#include <sstream>
+
+#include "base/kaldi-common.h"
+#include "util/common-utils.h"
+#include "hmm/posterior.h"
+#include "nnet3/nnet-example.h"
+#include "rnnlm/rnnlm-utils.h"
+#include "nnet3/nnet-diagnostics.h"
+
+using std::string;
```

Comment:
Per Google style guide,
"Do not use using-directives (e.g. using namespace foo)"
You may find it easier to put your code inside a namespace, e.g. kaldi::rnnlm or kaldi::nnet3.  Or make namespace rnnlm a sub-namespace of nnet3.